cmd/common/image_functions.go

```diff
@@ -198,95 +304,203 @@ func GetUntaggedManifests(ctx context.Context, acrClient api.AcrCLIClientInterfa
 			return nil, err
 		}
 	}
-	// Remove all manifests that should not be deleted
-	for i := 0; i < len(candidates); i++ {
-		if !ignoreList.Contains(*candidates[i].Digest) {
-			// if a manifest has no tags, is not part of a manifest list and can be deleted then it is added to the
-			// manifestsForCommand array.
-			if *(candidates[i].ChangeableAttributes).DeleteEnabled && *(candidates[i].ChangeableAttributes).WriteEnabled {
-				manifestsForCommand = append(manifestsForCommand, *candidates[i].Digest)
-				if dryRun && !ignoreReferrerManifests {
-					fmt.Printf("%s/%s@%s\n", loginURL, repoName, *candidates[i].Digest)
-				}
-			}
+
+	// Wait for all the goroutines to finish or return an error if one of them failed
+	if err := group.Wait(); err != nil {
+		return nil, err
+	}
+
+	// Remove everything from the candidates list that is in the ignoreList
+	for _, manifest := range candidates {
```

Comment:
why not just iterate through the ignore list and remove from the candidate list?

Reply:
Ultimately, we still need to read all of candidates since we actually output a list of digests, that said, the code isn't practical as it is deleting from the candidates map as we iterate through it and then iterates through what remains of the map a second time to build the list. Instead, I rolled it up into just one loop that does the check and adds the manifests to the list only if we do need to delete.